nmostesting/suites/BCP0050201Test.py

```diff
@@ -0,0 +1,474 @@
+
+# Generic capabilities from any namespace
+
+
+def cap_without_namespace(s):
+    match = re.search(r'^urn:[a-z0-9][a-z0-9-]+:cap:(.*)', s)
+    return match.group(1) if match else None
+
+
+def get_key_value(obj, name):
+    regex = re.compile(r'^urn:[a-z0-9][a-z0-9-]+:' + name + r'$')
+    for key, value in obj.items():
+        if regex.fullmatch(key):
+            return value
+    return obj[name]  # final try without a namespace
+
+
+def has_key(obj, name):
+    regex = re.compile(r'^urn:[a-z0-9][a-z0-9-]+:' + name + r'$')
+    for key in obj.keys():
+        if regex.fullmatch(key):
+            return True
+    return name in obj  # final try without a namespace
+
+
+class BCP0050201Test(GenericTest):
+    """
+    Runs Node Tests covering sender and receiver with IPMX/HKEP
+    """
+
+    def __init__(self, apis, **kwargs):
+        # Don't auto-test /transportfile as it is permitted to generate a 404 when master_enable is false
+        omit_paths = [
+            "/single/senders/{senderId}/transportfile",
+            "/single/senders/{senderId}/staged",
+            "/single/senders/{senderId}/active",
+            "/single/senders/{senderId}/constraints",
+            "/single/senders/{senderId}/transporttype",
+            "/single/receivers/{receiverId}/staged",
+            "/single/receivers/{receiverId}/active",
+            "/single/receivers/{receiverId}/constraints",
+            "/single/receivers/{receiverId}/transporttype",
+        ]
+        GenericTest.__init__(self, apis, omit_paths, **kwargs)
+        self.node_url = self.apis[NODE_API_KEY]["url"]
+        self.connection_url = self.apis[CONNECTION_API_KEY]["url"]
+        self.is04_resources = {"senders": {}, "receivers": {}, "_requested": [], "sources": {}, "flows": {}}
+        self.is05_resources = {"senders": [], "receivers": [],
+                               "_requested": [], "transport_types": {}, "transport_files": {}}
+        self.is04_utils = IS04Utils(self.node_url)
+        self.is05_utils = IS05Utils(self.connection_url)
+
+    # Utility function from IS0502Test
+    def get_is04_resources(self, resource_type):
+        """Retrieve all Senders or Receivers from a Node API, keeping hold of the returned objects"""
+        assert resource_type in ["senders", "receivers", "sources", "flows"]
+
+        # Prevent this being executed twice in one test run
+        if resource_type in self.is04_resources["_requested"]:
+            return True, ""
+
+        path_url = resource_type
+        full_url = self.node_url + path_url
+        valid, resources = self.do_request("GET", full_url)
+        if not valid:
+            return False, "Node API did not respond as expected: {}".format(resources)
+        schema = self.get_schema(NODE_API_KEY, "GET", "/" + path_url, resources.status_code)
+        valid, message = self.check_response(schema, "GET", resources)
+        if not valid:
+            raise NMOSTestException(message)
```

Comment:
See comment in #874  (probably return a tuple as you can't raise an `NMOSTestException` without the `test` object)